go-runtime/ftl/reflection/database.go

```diff
@@ -0,0 +1,55 @@
+package reflection
+
+import (
+	"database/sql"
+	"fmt"
+	"reflect"
+
+	"github.com/alecthomas/types/once"
+)
+
+type DBType int
+
+const (
+	DBTypePostgres DBType = iota
+)
+
+type ReflectedDatabaseHandle struct {
+	DBType DBType
+	DB     *once.Handle[*sql.DB]
+
+	// configs
+	Name string
+}
+
+func Database[T any](init func(ref Ref) *ReflectedDatabaseHandle) Registree {
+	ref := Ref{
+		Module: moduleForType(reflect.TypeFor[T]()),
+		Name:   getDatabaseName[T](),
+	}
+	return func(t *TypeRegistry) {
+		t.databases[ref] = init(ref)
+	}
+}
+
+func getDatabaseName[T any]() string {
```

Comment:
Don't we also extract this statically? If we have it statically, wouldn't it be simpler to just register it with the statically extracted name rather than having to do this at runtime?

Reply:
true, we can register it with the statically extracted name but I think we'll still need this for lookups using the type?